internal/util/ssa/matchers.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ssa
+
+import (
+	"fmt"
+
+	"github.com/onsi/gomega/types"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ManagedFieldMatcher is a gomega Matcher to check if a ManagedFieldEntry has the provider name and operation.
+func ManagedFieldMatcher(manager string, operation metav1.ManagedFieldsOperationType) types.GomegaMatcher {
```

Comment:
Should this be called MatchManagedFieldEntry?

(so we end up with `ContainElement(ssa.MatchManagedFieldEntry("manager", metav1.ManagedFieldsOperationUpdate)),`)